modules/samples/firealarm/component/api/src/main/java/org/homeautomation/firealarm/api/DeviceTypeServiceImpl.java

```diff
@@ -0,0 +1,431 @@
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentHashMap;
+
+
+/**
+ * This is the API which is used to control and manage device type functionality
+ */
+@SuppressWarnings("NonJaxWsWebServices")
+@API(name = "firealarm", version = "1.0.0", context = "/firealarm", tags = "firealarm")
+@DeviceType(value = "firealarm")
+public class DeviceTypeServiceImpl implements DeviceTypeService {
+
+    private static final String KEY_TYPE = "PRODUCTION";
+    private static Log log = LogFactory.getLog(DeviceTypeService.class);
+    private static ApiApplicationKey apiApplicationKey;
+    private ConcurrentHashMap<String, DeviceJSON> deviceToIpMap = new ConcurrentHashMap<>();
+
+    private static String shortUUID() {
+        UUID uuid = UUID.randomUUID();
+        long l = ByteBuffer.wrap(uuid.toString().getBytes(StandardCharsets.UTF_8)).getLong();
+        return Long.toString(l, Character.MAX_RADIX);
+    }
+
+    /**
+     * @param agentInfo device owner,id
+     * @return true if device instance is added to map
+     */
+    @Path("device/register")
+    @POST
+    @Consumes(MediaType.APPLICATION_JSON)
+    public Response registerDevice(final DeviceJSON agentInfo) {
+        String deviceId = agentInfo.deviceId;
+        if ((agentInfo.deviceId != null) && (agentInfo.owner != null)) {
+            deviceToIpMap.put(deviceId, agentInfo);
+            return Response.status(Response.Status.OK).build();
+        }
+        return Response.status(Response.Status.NOT_ACCEPTABLE).build();
+    }
+
+    /**
+     * @param deviceId unique identifier for given device type instance
+     * @param state    change status of sensor: on/off
+     */
+    @Path("device/{deviceId}/change-status")
+    @POST
+    @Feature(code = "change-status", name = "Change status of sensor: on/off",
+            description = "Change status of sensor: on/off")
+    public Response changeStatus(@PathParam("deviceId") String deviceId,
+                                 @QueryParam("state") String state,
+                                 @Context HttpServletResponse response) {
+        try {
+            if (!APIUtil.getDeviceAccessAuthorizationService().isUserAuthorized(new DeviceIdentifier(deviceId,
+                                                                                                     DeviceTypeConstants.DEVICE_TYPE))) {
+                return Response.status(Response.Status.UNAUTHORIZED.getStatusCode()).build();
+            }
+            String sensorState = state.toUpperCase();
```

Comment:
Ack, This is a generated code with iot device type maven tooling. Hence need to fix formatting in the tooling.